src/system/timer_data.hh

```diff
@@ -0,0 +1,145 @@
+/*
```

Comment:
After simplifications I suggest we can make two independent implementations of this class and just 
have one conditional compilation in the Timer class containing include of the right implementation.